.github/actions/bumpVersion/bumpVersion.js

```diff
@@ -2,33 +2,120 @@ const {exec} = require('child_process');
 // https://docs.github.com/en/free-pro-team@latest/actions/reference/environment-variables#default-environment-variables
 const [repoOwner, repoName] = process.env.GITHUB_REPOSITORY.split('/');
 
+/**
+ * Pad a number to be three digits (with leading zeros if necessary).
+ *
+ * @param {Number} number - Must be an integer.
+ * @returns {String} - A string representation of the number w/ length 3.
+ */
+function padToThreeDigits(number) {
+    if (number >= 100) {
+        return number.toString();
+    }
+    if (number >= 10) {
+        return `0${number.toString()}`;
+    }
+    return `00${number.toString()}`;
+}
+
+/**
+ * Generate the 12-digit versionCode for android.
+ * This version code allocates three digits each for MAJOR, MINOR, PATCH, and BUILD versions.
+ * As a result, our max version is 999.999.999-999.
+ *
+ * @param {String} npmVersion
+ * @returns {String}
+ */
+function generateAndroidVersionCode(npmVersion) {
+    return ''.concat(
+        padToThreeDigits(getMajorVersion(npmVersion)),
+        padToThreeDigits(getMinorVersion(npmVersion)),
+        padToThreeDigits(getPatchVersion(npmVersion)),
+        padToThreeDigits(getBuildVersion(npmVersion)),
+    );
+}
+
+/**
+ * Use the react-native-version package to update the version of a native platform.
+ *
+ * @param {String} platform – "ios" or "android"
+ * @param {String} versionCode – The version code to update the native platform.
+ */
+function updateNativeVersion(platform, versionCode) {
+    if (platform !== 'android' || platform !== 'ios') {
+        console.error('Invalid native platform specified!');
+        core.setFailed();
+    }
+
+    reactNativeVersion({
+        target: platform,
+        setBuild: versionCode,
+        amend: true,
+    })
+        .then(() => {
+            console.log(`Successfully updated ${platform} version to ${versionCode}`);
+        })
+        .catch((err) => {
+            console.log('Error updating native version:', `platform: ${platform}`, `versionCode: ${versionCode}`, err);
+            core.setFailed(err);
+        });
+}
+
+/**
+ * A callback function for a successful `npm version` command.
+ *
+ * @param {String} newVersion
+ */
+function postVersionUpdateNative(newVersion) {
+    const cleanNewVersion = semverClean(newVersion);
+    console.log(`Updated npm version to ${cleanNewVersion}! Updating native versions...`);
+
+    // The native version needs to be different on Android v.s. iOS
+    const androidVersionCode = generateAndroidVersionCode(cleanNewVersion);
+    console.log('Updating android:', `buildCode: ${androidVersionCode}`, `buildName: ${cleanNewVersion}`);
+    updateNativeVersion('android', generateAndroidVersionCode(cleanNewVersion));
+    console.log(`Updating ios to ${cleanNewVersion}`);
+    updateNativeVersion('ios', cleanNewVersion);
+}
+
 const MAX_RETRIES = 10;
 let errCount = 0;
 let shouldRetry;
 
 do {
     shouldRetry = false;
+    // eslint-disable-next-line no-loop-func
```

Comment:
Do we feel comfortable suppressing this lint error?

Reply:
Is there any way around it? It seems like it's necessary, right?

Reply:
Yeah, I think it's necessary in this case.